### ISSUE TITLE
lint: fix nestingReduce false positives

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#404]
`nestingReduce` checker seem to have false positives.
Running it over GOROOT reveals some unclear cases.
Checker suggests that nesting can be reduced, but it seems like a false positive in some cases.
More over, it doesn't give a hint how to do it.

Example for `negative_tests.go`:
```go

func noWarnings5(a int) {
	if true {
		if a == 5 {
			a++
			a++
			a++
			a++
			a++
			a++
		}
		a++
		a++
		a++
		a++
		a++
		a++
		a++
	}
}
```

**Comment by Quasilyte** [comment]
Insert a joke about `404` here.

**Comment by cristaloleg** [comment]
No issue has found, so there is no problem, gone fishing.
